mono/metadata/mono-perfcounters.c

```diff
@@ -1417,45 +1428,43 @@ mono_perfcounter_category_del (MonoString *name)
 	return TRUE;
 }
 
-/* this is an icall */
-MonoString*
-mono_perfcounter_category_help (MonoString *category, MonoString *machine)
+MonoStringHandle
+ves_icall_System_Diagnostics_PerformanceCounterCategory_CategoryHelpInternal (
+	const gunichar2* category, int category_length, MonoError *error)
+// previously mono_perfcounter_category_help
 {
-	ERROR_DECL (error);
-	MonoString *result = NULL;
+	TYPED_HANDLE_NAME (MonoString) result;
 	const CategoryDesc *cdesc;
-	error_init (error);
-	/* no support for counters on other machines */
-	if (mono_string_compare_ascii (machine, "."))
-		return NULL;
-	cdesc = find_category (category);
+	cdesc = find_category (category, category_length);
 	if (!cdesc) {
 		SharedCategory *scat = find_custom_category (category);
 		if (!scat)
-			return NULL;
-		result = mono_string_new_checked (mono_domain_get (), custom_category_help (scat), error);
-		if (mono_error_set_pending_exception (error))
-			return NULL;
-		return result;
+			goto return_null;
+		result = mono_string_new_handle (mono_domain_get (), custom_category_help (scat), error);
+		goto_if_nok (error, return_null);
+		goto exit;
```

Comment:
Why the `return result` to `goto exit`? What does it bring out of an extra line damage and an extra code path?

Reply:
The structuring with goto makes the correspondence to the previous code much clearer, easier to read diff. Esp. with goto return_null, but also goto exit.